xclim/indices/generic.py

```diff
@@ -496,8 +504,39 @@ def statistics(data: xr.DataArray, reducer: str, freq: str) -> xr.DataArray:
     return out
 
 
+def compare_operation(
+    left: xr.DataArray, op: str, right: xr.DataArray, constrain: Sequence[str] | None
+) -> xr.DataArray:
```

Comment:
This function seems a good idea indeed. I only have a suggestion for its name.
```suggestion
def compare_arrays(
    left: xr.DataArray, op: str, right: xr.DataArray, constrain: Sequence[str] | None
) -> xr.DataArray:
```